demos/guide/quic-hq-interop-server.c

```diff
@@ -0,0 +1,680 @@
+            bytes_written = 0;
+            if (bytes_read == 0)
+                break;
+        }
+    }
+
+done:
+    if (!SSL_stream_conclude(stream, 0))
+        fprintf(stderr, "Failed to conclude stream\n");
+
+out:
+    BIO_free(readbio);
+    return;
+}
+
+/**
+ * @brief Runs the QUIC server to accept and handle client connections.
+ *
+ * This function initializes a QUIC listener, binds it to the provided UDP
+ * socket, and enters a loop to accept client connections and process incoming
+ * QUIC streams. Each connection is handled until termination, and streams are
+ * processed individually using the `process_new_stream` function.
+ *
+ * @param ctx Pointer to the SSL_CTX object configured for QUIC.
+ * @param sock  BIO of the bound UDP socket.
+ *
+ * @return Returns 0 on error; otherwise, the server runs indefinitely.
+ *
+ * Operation:
+ * - Creates a QUIC listener using the provided SSL_CTX and associates it
+ *   with the specified UDP socket.
+ * - Waits for incoming QUIC connections and accepts them.
+ * - For each connection:
+ *   - Accepts incoming streams.
+ *   - Processes each stream using `process_new_stream`.
+ *   - Shuts down the connection upon completion.
+ *
+ * Error Handling:
+ * - If listener creation or connection acceptance fails, the function logs
+ *   an error message and exits the loop.
+ * - Cleans up allocated resources (e.g., listener, connection) on failure.
+ *
+ * Usage:
+ * - Call this function in the main server loop after setting up the
+ *   SSL_CTX and binding a UDP socket.
+ *
+ * Notes:
+ * - Uses blocking operations for listener, connection, and stream handling.
+ * - Incoming streams are processed based on the configured stream policy.
+ * - The server runs in an infinite loop unless a fatal error occurs.
+ */
+static int run_quic_server(SSL_CTX *ctx, BIO *sock)
+{
+    int ok = 0;
+    SSL *listener, *conn, *stream;
+    unsigned long errcode;
+
+    /*
+     * Create a new QUIC listener. Listeners, and other QUIC objects, default
+     * to operating in blocking mode. The configured behaviour is inherited by
+     * child objects.
+     */
+    if ((listener = SSL_new_listener(ctx, 0)) == NULL)
+        goto err;
+
+    /* Provide the listener with our UDP socket. */
+    SSL_set_bio(listener, sock, sock);
+
+    /* Begin listening. */
+    if (!SSL_listen(listener))
+        goto err;
+
+    /*
+     * Begin an infinite loop of listening for connections. We will only
+     * exit this loop if we encounter an error.
+     */
+    for (;;) {
+        /* Pristine error stack for each new connection */
+        ERR_clear_error();
+
+        /* Block while waiting for a client connection */
+        printf("Waiting for connection\n");
+        conn = SSL_accept_connection(listener, 0);
+        if (conn == NULL) {
+            fprintf(stderr, "error while accepting connection\n");
+            goto err;
+        }
+        printf("Accepted new connection\n");
```

Comment:
I would also suggest to add `SSL_DEFAULT_STREAM_MODE_NONE` flag here.

I think it's worth a comment saying we must explicitly tell openssl library the application is expecting to accept inbound streams from remote clients.

Reply:
I'm not sure I follow here.  The next line after "Accepted new connection" sets the accepted connection to automatically accept new incoming streams on the connection.  Is that not what you are suggesting above?

Reply:
I was just thinking to use this source code to demonstrate difference between default stream mode and multistream mode as described in quic-ssl(7). but I don't insist on this.

Reply:
added a comment for this